js/model.js

```diff
@@ -198,12 +199,12 @@ define([
         var isNotAvailable = !page.get('_isAvailable');
         if (isNotAvailable) continue;
 
-        if (!hasFoundCurrentPage && page.get('_id') === currentPageId) {
-          hasFoundCurrentPage = true;
+        if (!hasFoundCurrentPage) {
+          hasFoundCurrentPage = page.get('_id') === currentPageId;
           continue;
         }
 
-        if (hasFoundCurrentPage) {
+        if (!this.get('_shouldSkipOptionalPages') || !page.get('_isOptional')) {
```

Comment:
_possibly_ worth introducing a `skipOptionalPage` method so that you can replace these three very similar bits of code with:
```js
if (!this.skipOptionalPage(page))
```

Reply:
or `shouldSkipPage` - on the assumption that we _might_ add in more forms of 'page skipping' in future?